src/Loop.php

```diff
@@ -42,12 +42,11 @@ public static function setFactory(DriverFactory $factory = null)
         }
 
         self::$factory = $factory;
+        self::$driver = null; // reset it here, it will be actually instantiated inside execute() or get()
 
         if ($factory === null) {
-            self::$driver = null;
             self::$registry = null;
         } else {
-            self::$driver = self::createDriver();
             self::$registry = [];
```

Comment:
This issue isn't dependent on `wait()` specifically, but rather generically you reuse an already running event loop.

E.g.:
```php
$loop = new DriverInstanceFromSomeWhere;
Loop::execute(function() {
    Loop::storeState("foo", 1);
}, $loop);
Loop::execute(function() {
    var_dump(Loop::fetchState("foo")); // ugh???
}, $loop);
```